File: html/old/R2/SACO R2 2007 (Solutions)/digitsums.java
//class digitsum
//Problem "Digit Sum" for SACO 2nd round 2007
//Solution by Shen Tian

import java.io.*;

public class digitsums {
  public static void main(String[] args) throws IOException{
    BufferedReader in = new BufferedReader(new InputStreamReader(System.in));
    int N = Integer.parseInt(in.readLine());
    System.out.println(sum(N));
  }
  
  public static int sum(int x){
    if (x<10)
      return x;
     else
      return sum(x/10 + x%10);
  }
}

File: html/old/R2/SACO R2 2007 (Solutions)/digitsums.pas
{
   Solution for 'Digit Sums'
   SACO Problem 3 - Junior
   and SACO Problem 1 - Senior
   Solution by Ben Steenhuisen
}
program digitsums;
uses crt;
var inp,x:integer;
    outstr:string;
procedure digit(a:integer);
var tst:string;
    value,sum,io:integer;
begin
    str(a,tst);
    sum:=0;
    for x:=1 to length(tst) do begin
       val(tst[x],value,io);
       sum:=sum+value;
    end;
    if (sum>9) then digit(sum) else
    begin
        str(sum,tst);
        outstr:=tst;
    end;
end;
begin
    clrscr;
    readln(inp);
    digit(inp);
    writeln(outstr);
end.
